tests/performance/test_performance.py

```diff
@@ -54,15 +63,177 @@ def test_performance(config_path):
 
-    out = profile_transformer(transformer, dataset_gen, config['transform_size'],
-                              config['fit_size'])
+    out = profile_transformer(
+        transformer=transformer_instance,
+        dataset_generator=dataset_generator,
+        fit_size=config['fit_size'],
+        transform_size=config['transform_size'],
+    )
 
     assert out['Fit Time'] < config['expected']['fit']['time']
     assert out['Fit Memory'] < config['expected']['fit']['memory']
     assert out['Transform Time'] < config['expected']['transform']['time']
     assert out['Transform Memory'] < config['expected']['transform']['memory']
     assert out['Reverse Transform Time'] < config['expected']['reverse_transform']['time']
     assert out['Reverse Transform Memory'] < config['expected']['reverse_transform']['memory']
+
+
+def _round_to_magnitude(value):
+    for digits in range(-15, 15):
+        rounded = np.round(value, digits)
+        if rounded != 0:
+            return rounded
+
+    # We should never reach this line
+    raise ValueError("Value is too big")
+
+
+def find_transformer_boundaries(transformer, dataset_generator, fit_size,
+                                transform_size, iterations=1):
+    """Helper function to find valid candidate boundaries for performance tests.
+
+    The function works by:
+        - Running the profiling multiple times
+        - Averaging out the values for each metric
+        - Multiplying the found values by 10 (next order of magnitude).
+        - Rounding to the found order of magnitude
+
+    As an example, if a method took 0.012 seconds to run, the expected output
+    threshold will be set to 0.1, but if it took 0.016, it will be set to 0.2.
+
+    Args:
+        transformer (Transformer):
+            Transformer instance to profile.
+        dataset_generator (type):
+            Dataset Generator class to use.
+        fit_size (int):
+            Number of values to use when fitting the transformer.
+        transform_size (int):
+            Number of values to use when transforming and reverse transforming.
+        iterations (int):
+            Number of iterations to perform.
+
+    Returns:
+        pd.Series:
+            Candidate values for each metric.
+    """
+    results = [
+        profile_transformer(transformer, dataset_generator, transform_size, fit_size)
+        for _ in range(iterations)
+    ]
+    means = pd.DataFrame(results).mean(axis=0)
+    return (means * 10).apply(_round_to_magnitude)
+
+
+def make_test_case_config(transformer_class, transformer_kwargs, dataset_generator,
+                          fit_size, transform_size, iterations=1, output_path=None,
+                          config_name=None):
+    """Create a Test Case JSON file for the indicated transformer and dataset.
+
+    If output path is not given, the test case is created with the filename
+    ``{dataset_generator}_{fit_size}_{transform_size}.json`` inside the folder
```

Comment:
I think we should update this to `{config}_{dataset_generator}_{fit_size}_{transform_size}.json`

Reply:
Good catch